app/lib/core/models/test_history.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+
+class HistoryRecord {
+  final String title;
+  final String userId;
+  final String documentId;
+  HistoryRecord({
+    @required this.title,
+    @required this.userId,
+    this.documentId,
+  });
+
+  Map<String, dynamic> toJson() {
+    return {
+      'userId': userId,
+      'title': title,
+    };
+  }
+
+  static HistoryRecord fromFirestore(
+      Map<String, dynamic> data, String documentId) {
+    if (data == null) return null;
```

Comment:
No nulls please. We should return an `Optional` if it could be null.

- [ ] Make return type `Optional<HistoryRecord>`.

Reply:
Also, this should not be a `static` method. Let's add this as a named constructor `HistoryRecord.fromFirestore`.

Reply:
I was just testing another way to read data from firestore, I intended to change this but I completely forgot about it. I will change this in my next commit